### PR TITLE
feat!: enhanced types

### PR DESCRIPTION
**Pull request by angelxmoreno**
- created AdapterSearchOptions for unified search params
- created NewsArticleResponse to add pagination responses
- extended NewsArticleInterface properties

## Summary by CodeRabbit

- New Features
  - Unified pagination across sources (page/limit/offset/cursor) with responses including articles, totalCount, hasMore, and nextPage/nextCursor.

- Improvements
  - Articles now include optional description, imageUrl, category, language, and RSS author/media when available.
  - Aggregator returns a consolidated paginated payload and coordinates per-source pagination.
  - RSS parsing supports additional fields and enclosures.

- Documentation
  - Updated pagination and article-field docs; added roadmap/setup notes for a Currents News API adapter.

- Tests
  - Expanded adapter and aggregation tests covering pagination and edge cases.

### CODE REVIEW

File: src/adapters/NewsApiOrgAdapter.ts [hunk 1]
```diff
@@ -47,19 +52,39 @@ export class NewsApiOrgAdapter implements AdapterInterface {
...
     }
 
-    async getArticles(): Promise<NewsArticleInterface[]> {
+    async getArticles(options?: AdapterSearchOptions): Promise<NewsArticleResponse> {
+        // Merge runtime options with constructor search params
+        const params = {
+            ...this.searchParams,
+            ...(options?.page && { page: options.page }),
+            ...(options?.limit && { pageSize: Math.min(options.limit, 100) }), // NewsAPI max is 100
+        };
```
[the remark this answers is not in the document]
Comment (angelxmoreno) [reply src/adapters/NewsApiOrgAdapter.ts:61]: I agree the validation concerns are valid, but I think
  they should be addressed at the individual adapter level
   rather than trying to create generic pagination logic.
  Each API has different constraints - NewsAPI's 100-item
  limit doesn't apply to other APIs. Each adapter should
  validate according to its underlying API's specific
  requirements.

  Regarding offset conversion, I still think that adds
  unnecessary complexity when each adapter should work
  naturally with its API's pagination model.

File: src/adapters/RssAdapter.ts [hunk 2]
```diff
@@ -17,22 +19,45 @@ export class RssAdapter implements AdapterInterface {
...
+
+        return {
+            articles: paginatedArticles,
+            totalCount: newsArticles.length,
+            hasMore: paginatedArticles.length < newsArticles.length,
+            nextPage:
+                options?.page && options?.limit && paginatedArticles.length === options.limit
+                    ? options.page + 1
+                    : undefined,
+        };
```
[the remark this answers is not in the document]
Comment (angelxmoreno) [reply src/adapters/RssAdapter.ts]: This comment identifies real bugs in the RssAdapter's
  client-side pagination logic - especially the zero
  offset issue and inconsistent hasMore calculations.
  These should be fixed within the RssAdapter since it's
  handling client-side pagination of the full dataset.
